### ISSUE TITLE
Allow Lighthouse to reuse authed sessions from scripts

### ISSUE THREAD

**Issue by Friss** [WPO-Foundation/wptagent#202]
Currently we are testing our apps via Webpagetest and have a script that logs in and then navigates to the app. This works great for base WPT but in Lighthouse all of our scores are based on the login page and not the end app. 

Similar to https://github.com/WPO-Foundation/wptagent/issues/167 as far as passing cookies along but this is a slightly different end goal.

**Comment by pmeenan** [comment]
Thinking through it a bit more we could probably make it automatic if the WPT run was a script with more than one step.

**Comment by Friss** [comment]
Yeah would be nice if it was automatic and just started working for long scripts. 

I can also see the appeal of an option if you wanted to carry forward any cookies set by JavaScript from just a regular page load.

**Comment by sovanyio** [comment]
Piggybacking on this issue to potentially support broader auth implementations. I have a series of test cases that are currently scripted through node with lighthouse to support auth based on cookies + localStorage.

The script disables caching on the puppeteer instance and sets the clearStorage flag in lighthouse to false--in order to retain auth but simulate unprimed cache.

It might be beneficial to expose a scripting input for node-lighthouse in addition to the existing lighthouse support, in order to support more complex workflows. Would really love to get this working through WPT and standardized wrt machine power and resource variability.

Short of that, it would be cool if the options configuration object was exposed and configurable :)
